lnwallet/channel.go

```diff
@@ -3470,44 +3425,37 @@ func newHtlcResolution(signer Signer, localChanCfg *channeldb.ChannelConfig,
 	// transaction creates so we can generate the signDesc required to
 	// complete the claim process after a delay period.
 	htlcSweepScript, err := secondLevelHtlcScript(
-		revokeKey, delayKey, csvDelay,
+		keys.revocationKey, keys.delayKey, csvDelay,
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	return &OutgoingHtlcResolution{
+	// TODO: Signing with the delay key is wrong for remote commitments
+	localDelayTweak := SingleTweakBytes(keys.point, localChanCfg.DelayBasePoint)
+	resolution := OutgoingHtlcResolution{
 		Expiry:          htlc.RefundTimeout,
 		SignedTimeoutTx: timeoutTx,
 		SweepSignDesc: SignDescriptor{
 			PubKey:        localChanCfg.DelayBasePoint,
-			SingleTweak:   commitTweak,
+			SingleTweak:   localDelayTweak,
 			WitnessScript: htlcSweepScript,
 			Output: &wire.TxOut{
 				Value: int64(secondLevelOutputAmt),
 			},
 			HashType: txscript.SigHashAll,
 		},
-	}, nil
+	}
```

Comment:
Any particular reason for the added extra line? We can simply return the `OutgoingHtlcResolution` directly, w/o creating a temporary variable whose lifetime is a single line.

Reply:
It looked awkward returning multiple values where one is a multiline value. I can change back if you prefer, but it reads better to me this way.